src/components/Explore/TracesByService/Tabs/Structure/StructureScene.tsx

```diff
@@ -55,34 +60,88 @@ export class StructureTabScene extends SceneObjectBase<ServicesTabSceneState> {
       ...state,
     });
 
-    this.addActivationHandler(this._onActivate.bind(this));
+    this.addActivationHandler(() => {
+      this._onActivate();
+      return () => this._onDeactivate();
+    });
+  }
+
+  private _onDeactivate() {
+    if (this.updateTimeoutId) {
+      clearTimeout(this.updateTimeoutId);
+      this.updateTimeoutId = null;
+    }
+    this.lastProcessedData = null;
   }
 
   public _onActivate() {
     this.state.$data?.subscribeToState((state) => {
-      this.setState({ loading: state.data?.state === LoadingState.Loading });
-
+      if (state.data?.state === LoadingState.Loading) {
+        this.setState({ loading: true });
+        return;
+      }
+      
       if (
         (state.data?.state === LoadingState.Done || state.data?.state === LoadingState.Streaming) &&
         state.data?.series.length
       ) {
         const frame = state.data?.series[0].fields[0].values[0];
         if (frame) {
-          const response = JSON.parse(frame) as TraceSearchMetadata[];
-          const merged = mergeTraces(response);
-          merged.children.sort((a, b) => countSpans(b) - countSpans(a));
-          this.setState({
-            tree: merged,
-            panel: new SceneFlexLayout({
-              height: '100%',
-              wrap: 'wrap',
-              children: this.getPanels(merged),
-            }),
-          });
+          const now = Date.now();
+          // Prevent duplicate processing of the same data
+          // Skip if we processed this exact data less than 100ms ago
```

Comment:
> Skip if we processed this exact data less than 100ms ago

If it's exactly the same data do we need to do any processing at all or could be skip it altogether? 🤔

Reply:
This is returned immediately if it's duplicate data though 🤔